pt/lessons/basics/strings.md

```diff
@@ -0,0 +1,151 @@
+O módulo String já fornece dois métodos para obtê-los, `graphemes/1` e `codepoints/1`. Vamos ver um exemplo:
+
+```elixir
+iex> string = "\u0061\u0301"
+"á"
+
+iex> String.codepoints string
+["a", "́"]
+
+iex> String.graphemes string
+["á"]
+```
+
+## Funções de String
+
+Vamos rever duas das funções mais importantes e úteis do módulo String. Essa lição cobrirá apenas uma parte das funções disponíveis. Para ver a lista completa de funções visite a documentação oficial de [`String`](http://elixir-lang.org/docs/stable/elixir/String.html).
+
+### `length/1`
+
+Retorna o número de Graphemes na string.
+
+```elixir
+iex> String.length "Hello"
+5
+```
+
+### `replace/3`
+
+Retorna uma nova string substituindo um padrão atual por uma nova string de substituição.
+
+```elixir
+iex> String.replace("Hello", "e", "a")
+"Hallo"
+```
+
+### `duplicate/2`
+
+Retorna uma nova string repetida n vezes.
+
+```elixir
+iex> String.duplicate("Oh my ", 3)
+"Oh my Oh my Oh my "
+```
+
+### `split/2`
+
+Retorna uma lista de strings divididas por um padrão.
+
+```elixir
+iex> String.split("Hello World", " ")
+["Hello", "World"]
+```
+
+## Exercícios
+
+Vamos ver alguns exercícios simples para demonstrar que estamos prontos para trabalhar com Strings!
+
+### Anagramas
+
+A e B são considerados anagramas se há alguma forma de rearranjar A ou B para torná-las iguais. Por exemplo:
+
++ A = super
++ B = perus
+
+Se nós rearranjarmos os caracteres da string A, podemos obter a string B, e vice-versa.
+
+Então, como podemos checar se duas strings são Anagramas em Elixir? A solução mais fácil é apenas ordenar os graphemes de cada string em ordem alfabética e então checar se as duas listas são iguais. Vamos tentar:
+
+```elixir
+defmodule Anagram do
+  def anagrams?(a, b) when is_binary(a) and is_binary(b) do
+    sort_string(a) == sort_string(b)
+  end
+
+  def sort_string(string) do
+    string
+    |> String.downcase
+    |> String.graphemes
+    |> Enum.sort
+  end
+end
+```
+
+Primeiro vamos olhar para `anagrams?/2`. Estamos checando se os parâmetros que recebemos são binários ou não. Essa é a forma de checar se uma parâmetro é uma String em Elixir.
+
+Depois disso, estamos chamando a função que ordena as strings em ordem alfabética, primeiro deixando a string em letras minúsculas e então usando `String.graphemes`, que retorna a lista com os Graphemes da string. Bastante simples, não acha?
+
+Vamos checar a saída dessa função no iex:
+
+```elixir
+iex> Anagram.anagrams?("Hello", "ohell")
+true
+
+iex> Anagram.anagrams?("María", "íMara")
+true
+
+iex> Anagram.anagrams?(3, 5)
+** (FunctionClauseError) no function clause matching in Anagram.anagrams?/2
+    iex:2: Anagram.anagrams?(3, 5)
+```
+
+Como você pode ver, a última chamada a `anagrams?` causou um FunctionClauseError. Esse erro nos diz que não há nenhuma função no nosso módulo de acordo com o padrão de receber dois argumentos não binários, e isso é exatamente o que nós queremos, receber apenas duas strings e nada mais.
```

Comment:
There are other "checar" and variants in the paragraphs above...

Other than that, good work! 👍